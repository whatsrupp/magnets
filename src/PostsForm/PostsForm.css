.PostsForm{
    box-sizing: border-box;

    display: flex;
    flex-direction: column;
    padding: 20px;
    margin: 20px;
    
    border-radius: 5px;
    box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2) , 0 6px 20px 0 rgba(0,0,0,0.19) ;
}

.PostsForm__title-group, .PostsForm__contents-group {
    border-bottom: 2px solid;
    margin: 5px;
    display: flex;
    flex-basis: fill;
}

.PostsForm__contents-group{
    flex-direction: column;
    border-bottom: 0px;
}


.PostsForm__field{
    background-color: transparent;
    border: 0px solid;
    flex-grow: 15;
    overflow: hidden;
}
.PostsForm__field:focus{
    outline: none
}

.PostsForm__label{
    flex-grow: 1;
}

.PostsForm__submit-button{
    transition-duration: 0.4s;
    background-color: transparent;
    border: 0px;
    width: 100px;
    border: 1px solid;
    border-radius: 2px;
   
}

.PostsForm__submit-button:hover{
    background-color: orange;
    cursor: pointer;
}

.PostsForm__field--text-area{
    resize: none;
    overflow: auto;
    height: auto;
}

.PostsForm__controls{
    display: flex;
    flex-basis: fill;
    justify-content: flex-end;
    padding-top: 10px;
}