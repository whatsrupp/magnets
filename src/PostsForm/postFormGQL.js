import gql from 'graphql-tag';

export const createPostMutation = gql`
    mutation createPostMutation($title: String!, $content: String!) {
        createPost(title: $title, content: $content){
            _id
            title
            content
        }
    }
`
const getPostsQuery = gql`query TodoAppQuery {
    posts{
       title,
       content,
       _id
    }
   }`

const updateFunction = (cache, {data:{ createPost}})=>{
    const data = cache.readQuery({query: getPostsQuery})
    data.posts.push(createPost)
    cache.writeQuery({ query: getPostsQuery, data })
};

export const createPostOptions = {
    options: {
        update: updateFunction
    }
}