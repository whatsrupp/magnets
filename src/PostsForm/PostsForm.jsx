import React, { Component } from 'react'
import './PostsForm.css'

class PostsForm extends Component {
    constructor(props){
        super(props)
        this.state = {
            title: '',
            contents: ''
        }
        this.handleTitleChange = this.handleTitleChange.bind(this);
        this.handleContentsChange = this.handleContentsChange.bind(this);

    }

    handleTitleChange(event) {
        this.setState({title: event.target.value});
    }
    handleContentsChange(event) {
        this.setState({contents: event.target.value});
    }

    handleSubmit(event) {
        event.preventDefault()
    }

    render(){
        return(
            <form className="PostsForm"onSubmit={this.handleSubmit}>
                <div className="PostsForm__title-group" >
                    <label className="PostsForm__label">
                        Title
                    </label>
                    <input className="PostsForm__field"type="text" value={this.state.title} onChange={this.handleTitleChange} />
                </div>
                <div className="PostsForm__contents-group" >
                    <label className="PostsForm__label">
                        
                    </label>
                    <textarea className="PostsForm__field PostsForm__field--text-area"type="text-field" value={this.state.contents} onChange={this.handleContentsChange} />
                </div>
                <div className="PostsForm__controls">
                <input className="PostsForm__submit-button"type="submit" value="Submit" />
                </div>
            </form>
        )
    }
}

export default PostsForm
