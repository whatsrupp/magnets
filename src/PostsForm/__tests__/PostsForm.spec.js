import React, { Component } from 'react'
import { render } from 'enzyme';
import PostsForm from '../PostsForm';

describe('<PostsForm />', ()=>{
    it('renders', ()=>{
        const wrapper = render(<PostsForm />);
        expect(wrapper).toMatchSnapshot();
    })
});