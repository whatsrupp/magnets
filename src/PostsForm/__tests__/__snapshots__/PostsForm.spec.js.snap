// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsForm /> renders 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "PostsForm",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "PostsForm__input-group",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "PostsForm__label",
            },
            "children": Array [
              Object {
                "data": "Title",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "label",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "PostsForm__field",
                "type": "text",
                "value": "",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "PostsForm__field",
              "type": "text",
              "value": "",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "PostsForm__label",
              },
              "children": Array [
                Object {
                  "data": "Title",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "label",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "PostsForm__submit-button",
            "type": "submit",
            "value": "Submit",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "type": undefined,
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "type": undefined,
            "value": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "PostsForm__submit-button",
          "type": "submit",
          "value": "Submit",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "PostsForm__input-group",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "PostsForm__label",
              },
              "children": Array [
                Object {
                  "data": "Title",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "label",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "PostsForm__field",
                  "type": "text",
                  "value": "",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "PostsForm__field",
                "type": "text",
                "value": "",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "PostsForm__label",
                },
                "children": Array [
                  Object {
                    "data": "Title",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "label",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "type": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "type": undefined,
          "value": undefined,
        },
      },
    ],
    "name": "form",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
